File: python_baekjoon_구현/스택.py
n = int(input())     # 주어지는 명령의 수
commands = [input() for _ in range(n)]     # 주어지는 명령들
stack = []

for command in commands:
    if 'push' in command:
        stack.append(command.split()[-1])
    elif command == 'top':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])
    elif command == 'size':
        print(len(stack))
    elif command == 'empty':
        if len(stack) == 0:
            print(1)
        else:
            print(0)
    elif command == 'pop':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())

File: 백준/Silver/10828. 스택/스택.js
const array = require('fs').readFileSync('/dev/stdin').toString().split('\n');

const result = [];
const stack = [];

for (let i = 1; i < array.length; i++) {
  const command = array[i].split(" ");
  if (command[0] === "push") stack.push(command[1]);
  else if (command[0] === "top") {
      if (stack.length === 0) result.push('-1');
      else result.push(''+stack[stack.length - 1])
  }
  else if (command[0] === "size") result.push(''+stack.length);
  else if (command[0] === "empty") result.push(''+Number(!stack.length));
  else if (command[0] === "pop")
    if (stack.length === 0) result.push('-1')
    else result.push(stack.pop());
}

console.log(result.join('\n'));